projects/scream/scream_post/translate_prognostic_output.py

```diff
@@ -0,0 +1,297 @@
+    )
+    ds[
+        "shortwave_transmissivity_of_atmospheric_column"
+    ] = shortwave_transmissivity_of_atmospheric_column
+    ds[
+        "override_for_time_adjusted_total_sky_downward_longwave_flux_at_surface"
+    ] = override_for_time_adjusted_total_sky_downward_longwave_flux_at_surface
+    return ds
+
+
+def compute_prognostic_run_diagnostics(
+    ds: xr.DataArray, label: str, tendency_variables: [], hydrostatic: bool = False
+):
+    """
+    Compute diagnostics for a prognostic run
+
+    Args:
+        ds: xarray dataset containing prognostic run output
+        label: either "machine_learning" or "nudging"
+        tendency_variables: list of variables that are either nudged or ML corrected
+        hydrostatic: whether the run is hydrostatic
+    """
+    delp = ds[DELP]
+    temperature_tendency_name = f"{label}_{TEMP}_tend"
+    humidity_tendency_name = f"{label}_{SPHUM}_tend"
+    temperature_tendency = ds[temperature_tendency_name]
+    humidity_tendency = ds[humidity_tendency_name]
+    water_vapor_path = ds["VapWaterPath"]
+    # compute column-integrated diagnostics
+    if hydrostatic:
+        net_heating = vcm.column_integrated_heating_from_isobaric_transition(
+            temperature_tendency, delp, "lev"
+        )
+    else:
+        net_heating = vcm.column_integrated_heating_from_isochoric_transition(
+            temperature_tendency, delp, "lev"
+        )
+    diags = {
+        f"net_moistening_due_to_{label}": vcm.mass_integrate(
+            humidity_tendency, delp, dim="lev"
+        ).assign_attrs(
+            units="kg/m^2/s",
+            description=f"column integrated moisture tendency due to {label}",
+        ),
+        f"column_heating_due_to_{label}": net_heating.assign_attrs(
+            units="W/m^2"
+        ).assign_attrs(description=f"column integrated heating due to {label}"),
+        f"water_vapor_path": water_vapor_path.assign_attrs(units="kg/m^2").assign_attrs(
+            description=f"water vapor path"
+        ),
+    }
+    # add 3D tendencies to diagnostics
+    if label == "nudging":
+        diags_3d: Mapping[Hashable, xr.DataArray] = {
+            f"{k}_tendency_due_to_nudging": ds[f"{label}_{k}_tend"]
+            for k in tendency_variables
+        }
+    elif label == "machine_learning":
+        diags_3d = {
+            "dQ1": temperature_tendency.assign_attrs(units="K/s").assign_attrs(
+                description=f"air temperature tendency due to {label}"
+            ),
+            "dQ2": humidity_tendency.assign_attrs(units="kg/kg/s").assign_attrs(
+                description=f"specific humidity tendency due to {label}"
+            ),
+        }
+    diags.update(diags_3d)
+    if (
+        label == "machine_learning"
+        and "U" in tendency_variables
+        and "V" in tendency_variables
+    ):
+        diags.update(compute_ml_momentum_diagnostics(ds))
+    return xr.Dataset(diags)
+
+
+def compute_ml_momentum_diagnostics(ds: xr.DataArray):
```

Comment:
This is what I thought too, but I couldn't locate how the nudged runs use these variables.

Reply:
True, I guess for the nudged run the 3D momentum tendency diagnostics are covered by the `eastwared_wind_tendency_due_to_nudging` and `northward_wind_tendency_due_to_nudging` diagnostics; it's mainly the column integrated versions of those that are missing.

Since this is also an issue in the FV3GFS/SHiELD prognostic run, I think it's fine to punt on that for now.  We'd probably want to update both at the same time in a separate PR, so that we ensure we use the same naming conventions.